### ISSUE TITLE
Move twisted.mail.imap4 interfaces into a common interfaces file

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8808]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8808|
|Type|task|
|Created|2016-09-08 18:31:11Z|
|Branch|https://github.com/twisted/twisted/tree/8808-imap4-interfaces|

<details><summary>Searchable metadata</summary>

```
trac-id__8808 8808
type__task task
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__8808_imap4_interfaces 8808-imap4-interfaces
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1473359471934361 1473359471934361
changetime__1473367767839875 1473367767839875
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Okay, please review.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc set owner to @hawkowl|
|-|-|

https://github.com/twisted/twisted/pull/538#issuecomment-245730985

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

In changeset cc51187e5cde697df6ec408c799669a37d3bd130
```
#!CommitTicketReference repository="" revision="cc51187e5cde697df6ec408c799669a37d3bd130"
Merge 8808-imap4-interfaces: Split out imap4 interfaces

Author: hawkowl
Reviewer: rodrigc
Fixes: #8808
```
